### PR TITLE
Add loop contracts and harness for `run_utf8_validation`

### PR DESCRIPTION
**Pull request by qinheping**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Review by celinval — APPROVED** [review]

File: library/core/src/lib.rs [hunk 1]
```diff
@@ -230,6 +230,7 @@
 #![feature(unboxed_closures)]
 #![feature(unsized_fn_params)]
 #![feature(with_negative_coherence)]
+#![feature(proc_macro_hygiene)]
```
Comment (celinval) [inline library/core/src/lib.rs:233]: please add cfg_attr

File: library/core/src/str/validations.rs [hunk 2]
```diff
@@ -271,3 +277,28 @@ pub const fn utf8_char_width(b: u8) -> usize {
...
 /// Mask of the value bits of a continuation byte.
 const CONT_MASK: u8 = 0b0011_1111;
+
+#[cfg(kani)]
+#[unstable(feature = "kani", issue = "none")]
+pub mod verify {
+    use super::*;
+
+    #[kani::proof]
+    #[kani::unwind(8)]
```
Comment (zhassan-aws) [inline library/core/src/str/validations.rs:287]: Is the unwind necessary?

**Review by feliperodri — APPROVED** [review]
This is awesome. A bit more documentation on the bounds that still exist in the harness would be more helpful, but I couldn't find any problem with the loop invariants that you added, thus, not blocking the PR. Can't wait to see some performance results in Kani using these contracts while verifying code that use strings.

File: library/core/src/str/validations.rs (hunk 2 above)
Comment (qinheping) [reply library/core/src/str/validations.rs:287]: No, I think it is accidentally copied from another harness.

File: scripts/run-kani.sh [hunk 3]
```diff
@@ -183,7 +183,7 @@ main() {
 
     echo "Running Kani verify-std command..."
 
-    "$kani_path" verify-std -Z unstable-options ./library --target-dir "$temp_dir_target" -Z function-contracts -Z mem-predicates --output-format=terse $command_args
+    "$kani_path" verify-std -Z unstable-options ./library --target-dir "$temp_dir_target" -Z function-contracts -Z mem-predicates -Z loop-contracts --output-format=terse $command_args --enable-unstable --cbmc-args --object-bits 12
```
[the remark this answers is not in the document]
Comment (qinheping) [reply scripts/run-kani.sh:186]: Yes, DFCC create internal objects.

File: library/core/src/str/validations.rs [hunk 4]
```diff
@@ -271,3 +277,28 @@ pub const fn utf8_char_width(b: u8) -> usize {
...
+    pub fn check_run_utf8_validation() {
+        if kani::any() {
+            // TODO: ARR_SIZE can be `std::usize::MAX` with cbmc argument
+            // `--arrays-uf-always`
```
[the remark this answers is not in the document]
Comment (qinheping) [reply library/core/src/str/validations.rs:291]: We don't want to use it for all harness while we can't specify it for certain harnesses now.

File: library/core/src/str/validations.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (qinheping) [reply library/core/src/str/validations.rs:287]: It is accidentally copied from another harness. I will remove it.

**Review by celinval — COMMENTED** [review]
trigger approval check

**Pull request merged**